osu.Game.Rulesets.Catch/Skinning/Legacy/LegacyCatchComboCounter.cs

```diff
@@ -1,100 +1,132 @@
 
         public LegacyCatchComboCounter()
         {
-            AutoSizeAxes = Axes.Both;
+            PopOutCountText.Origin = PopOutCountText.Anchor = Anchor.Centre;
+            DisplayedCountText.Origin = DisplayedCountText.Anchor = Anchor.Centre;
+
+            Content.ChangeChildDepth(PopOutCountText, float.MinValue);
+            UsesFixedAnchor = true;
+        }
 
-            Alpha = 0f;
-            Anchor = Anchor.Centre;
-            Origin = Anchor.Centre;
-            Scale = new Vector2(0.8f);
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
 
-            InternalChildren = new Drawable[]
+            lastJudgementResult = gameplayState.LastJudgementResult.GetBoundCopy();
+            lastJudgementResult.BindValueChanged(result =>
             {
-                explosion = new LegacyRollingCounter(LegacyFont.Combo)
-                {
-                    Alpha = 0.65f,
-                    Blending = BlendingParameters.Additive,
-                    Anchor = Anchor.Centre,
-                    Origin = Anchor.Centre,
-                    Scale = new Vector2(1.5f),
-                },
-                counter = new LegacyRollingCounter(LegacyFont.Combo)
-                {
-                    Anchor = Anchor.Centre,
-                    Origin = Anchor.Centre,
-                },
-            };
-        }
+                if (!result.NewValue.Type.AffectsCombo() || !result.NewValue.HasResult)
+                    return;
 
-        private int lastDisplayedCombo;
+                if (!result.NewValue.IsHit)
+                    return;
 
-        public void UpdateCombo(int combo, Color4? hitObjectColour = null)
-        {
-            if (combo == lastDisplayedCombo)
-                return;
+                if (result.NewValue?.HitObject is IHasComboInformation catchObject)
+                    PopOutCountText.Colour = catchObject.GetComboColour(skin);
+            });
 
-            // There may still be existing transforms to the counter (including value change after 250ms),
-            // finish them immediately before new transforms.
-            counter.SetCountWithoutRolling(lastDisplayedCombo);
+            FinishTransforms(true);
+        }
 
-            lastDisplayedCombo = combo;
+        protected override void Update()
+        {
+            base.Update();
 
-            if ((Clock as IGameplayClock)?.IsRewinding == true)
+            if (drawableRuleset != null)
             {
-                // needs more work to make rewind somehow look good.
-                // basically we want the previous increment to play... or turning off RemoveCompletedTransforms (not feasible from a performance angle).
-                Hide();
-                return;
+                var catcher = ((CatchPlayfield)drawableRuleset.Playfield).Catcher;
+                X = Parent!.ToLocalSpace(catcher.ScreenSpaceDrawQuad.Centre).X;
```

Comment:
I do wonder about this. Maybe it's fine for now, but I do recall that people have asked for things to be easily attachable to the catcher ("things" being an accuracy counter or something). Due to that I'd be wondering at this stage whether we want to have a `SkinnableContainer` rooted at the catcher or something for that express purpose.

Probably not something that is required to be acted upon immediately for merge, just food for thought maybe.

Reply:
Pretty interesting suggestion. It can't be applied at the moment as the available skin layers are defined at a ruleset-agnostic level (see `GlobalSkinnableContainers`).